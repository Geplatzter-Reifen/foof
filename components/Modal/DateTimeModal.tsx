import { Modal, StyleSheet, View } from "react-native";
import { Button, Layout, Text } from "@ui-kitten/components";
import DateTimeButton from "@/components/Modal/DateTimeButton";
import { useState } from "react";

type DateTimeModalProps = {
  modalVisible: boolean;
  onClose: () => void;
  onSave: () => void;
  onStartDateChange: (startDate: Date) => void;
  onEndDateChange: (endDate: Date) => void;
  initialStartDate?: Date;
  initialEndDate?: Date;
};

function DateTimeModal(props: DateTimeModalProps) {
  const {
    modalVisible,
    onClose,
    onSave,
    onStartDateChange,
    onEndDateChange,
    initialStartDate,
    initialEndDate,
  } = props;

  const [startDate, setStartDate] = useState(initialStartDate || new Date());
  const [endDate, setEndDate] = useState(initialEndDate || new Date());

  const handleStartDateChange = (date: Date) => {
    onStartDateChange(date);
    setStartDate(date);
  };

  const handleEndDateChange = (date: Date) => {
    onEndDateChange(date);
    setEndDate(date);
  };

  return (
    <Modal
      animationType="fade"
      transparent={true}
      visible={modalVisible}
      statusBarTranslucent={true}
    >
      <View style={styles.centeredView}>
        <View style={styles.modalView}>
          <Text category="h5">Startzeit</Text>
          <Layout style={styles.row}>
            <DateTimeButton
              date={startDate}
              mode={"date"}
              onDateChange={handleStartDateChange}
            />
            <DateTimeButton
              date={startDate}
              mode={"time"}
              onDateChange={handleStartDateChange}
            />
          </Layout>
          <Text category="h5">Endzeit</Text>
          <Layout style={styles.row}>
            <DateTimeButton
              date={endDate}
              mode={"date"}
              onDateChange={handleEndDateChange}
            />
            <DateTimeButton
              date={endDate}
              mode={"time"}
              onDateChange={handleEndDateChange}
            />
          </Layout>
          <Layout style={styles.row}>
            <Button style={styles.button} onPress={onClose}>
              Abbrechen
            </Button>
            <Button style={styles.button} onPress={onSave}>
              Speichern
            </Button>
          </Layout>
        </View>
      </View>
    </Modal>
  );
}

const styles = StyleSheet.create({
  centeredView: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "rgba(0, 0, 0, 0.5)",
  },
  modalView: {
    margin: 20,
    backgroundColor: "white",
    borderRadius: 20,
    padding: 20,
    alignItems: "center",
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
  row: {
    flexDirection: "row", // Arrange inputs horizontally
    justifyContent: "space-between", // Space them out
    alignItems: "center", // Center items vertically
    marginBottom: 5, // Add spacing between rows
  },
  button: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
    margin: 5,
    alignSelf: "center",
  },
});

export default DateTimeModal;
