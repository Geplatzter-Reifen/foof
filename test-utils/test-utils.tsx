// https://testing-library.com/docs/react-testing-library/setup/#add-custom-queries
import React from "react";
import { render, RenderOptions } from "@testing-library/react-native";
import { FontAwesomeIconsPack } from "@/components/Font/fontAwesome";
import { mapping } from "@eva-design/eva";
import {
  IconRegistry,
  ApplicationProvider,
  ModalService,
} from "@ui-kitten/components";
import * as eva from "@eva-design/eva";
import { foofDarkTheme, foofLightTheme } from "@/constants/custom-theme";
import { library } from "@fortawesome/fontawesome-svg-core";
import { far } from "@fortawesome/free-regular-svg-icons";
import { fas } from "@fortawesome/free-solid-svg-icons";
import { fab } from "@fortawesome/free-brands-svg-icons";

// Iconpacks laden
library.add(far, fas, fab);

// ModalService wird verwendet, um die Modal-Positionierung zu steuern
ModalService.setShouldUseTopInsets = true;

type Props = {
  children: React.ReactNode;
  use_dark_theme?: boolean;
};
// Der Wrapper wird verwendet, um die Provider zu umschließen, die für die Komponenten notwendig sind
const AllTheProviders = ({ children, use_dark_theme = true }: Props) => {
  const theme = use_dark_theme
    ? { ...eva.dark, ...foofDarkTheme }
    : { ...eva.light, ...foofLightTheme };
  return (
    <>
      <IconRegistry icons={FontAwesomeIconsPack} />
      <ApplicationProvider {...eva} customMapping={mapping} theme={theme}>
        {children}
      </ApplicationProvider>
    </>
  );
};

// Erstellt einen Custom-Render, der die Provider um die Komponenten wickelt
const customRender = (
  component: React.ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
) => {
  return render(component, { wrapper: AllTheProviders, ...options });
};

// Exportiert alle Funktionen der Testing Library, weswegen man diese dann auch von dieser Datei importieren kann
export * from "@testing-library/react-native";
export { customRender as render };
