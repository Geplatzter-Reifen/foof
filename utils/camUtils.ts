import { getTourRoute } from "@/services/data/routeService";
import type { FeatureCollection } from "geojson";
import { Tour } from "@/database/model/model";
import { Camera } from "@rnmapbox/maps";
import React from "react";
import { calculateBounds } from "@/utils/locationUtils";

// Function to display the route on the map by adjusting the camera to fit the route's bounds
export function fitRouteInCam(
  activeTour: Tour | undefined,
  camera: React.RefObject<Camera>,
) {
  getTourRoute(activeTour?.id!).then((route) => {
    if (!route) {
      throw new Error("Keine Route importiert!");
    }
    const geoJSON: FeatureCollection = JSON.parse(route.geoJson);
    const bounds = calculateBounds(geoJSON);

    camera.current?.setCamera({
      bounds: bounds,
      padding: {
        paddingLeft: 30,
        paddingRight: 30,
        paddingTop: 30,
        paddingBottom: 150,
      },
      animationDuration: 2000,
      heading: 0,
      animationMode: "flyTo",
    });
  });
}
