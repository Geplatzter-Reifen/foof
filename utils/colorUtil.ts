export type RGB = {
  r: number;
  g: number;
  b: number;
};
export function hexToRgb(hex: string): RGB {
  hex = hex.replace(/^#/, "");

  const r = parseInt(hex.substring(0, 2), 16);
  const g = parseInt(hex.substring(2, 4), 16);
  const b = parseInt(hex.substring(4, 6), 16);
  if (isNaN(r) || isNaN(g) || isNaN(b)) {
    throw new Error("Invalid hex string");
  }

  return { r, g, b };
}

export function hexToRgba(hex: string, alpha: number) {
  const { r, g, b } = hexToRgb(hex);
  return `rgba(${r}, ${g}, ${b}, ${alpha})`;
}
