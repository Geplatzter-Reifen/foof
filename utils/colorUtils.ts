export function isHexValid(hex: string) {
  return /^#[0-9A-F]{6}$/i.test(hex);
}

export function hexToRgb(hex: string) {
  if (!isHexValid(hex)) {
    throw new Error("Invalid hex");
  }
  hex = hex.replace(/^#/, "");

  const r = parseInt(hex.substring(0, 2), 16);
  const g = parseInt(hex.substring(2, 4), 16);
  const b = parseInt(hex.substring(4, 6), 16);

  return { r, g, b };
}

export function hexToRgba(hex: string, alpha: number) {
  if (!isHexValid(hex)) {
    throw new Error("Invalid hex");
  }
  if (alpha < 0 || alpha > 1) {
    throw new Error("Invalid alpha");
  }
  const { r, g, b } = hexToRgb(hex);
  return `rgba(${r}, ${g}, ${b}, ${alpha})`;
}
