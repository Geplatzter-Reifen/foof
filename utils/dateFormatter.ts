import { format } from 'date-fns';

export const DATE: string = "dd.MM.yyyy"
export const TIME: string = "HH:mm"
export const TIME_UHR: string = "HH:mm Uhr"
export const DATE_TIME: string = "dd.MM.yyyy HH:mm"

export function dateFormat(date: Date, formatString: string): string {
  return format(date, formatString);
}