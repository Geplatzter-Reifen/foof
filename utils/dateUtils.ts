import { differenceInMilliseconds, format } from "date-fns";

export enum DateFormat {
  DATE = "dd.MM.yyyy",
  TIME = "HH:mm",
  DATE_TIME = "dd.MM.yyyy HH:mm",
}

export function formatDate(
  date: Date | number,
  formatString: DateFormat,
): string {
  if (typeof date === "number") {
    return format(new Date(date), formatString);
  }

  return format(date, formatString);
}

export function getDurationInMs(start: Date, end: Date) {
  return differenceInMilliseconds(end, start);
}

export function getDurationFormatted(start: Date, end: Date): string {
  if (end < start) {
    throw new Error("End date is before start date");
  }

  return getDurationMsFormatted(getDurationInMs(start, end));
}

export function getDurationMsFormatted(ms: number) {
  const totalMinutes = Math.floor(ms / 60000);

  const days = Math.floor(totalMinutes / 1440); // 1 Tag = 1440 Minuten
  const hours = Math.floor((totalMinutes % 1440) / 60);
  const minutes = totalMinutes % 60;
  let dayPart = "";
  let hoursPart = "";
  if (hours >= 1) {
    hoursPart = `${hours}h `;
  }
  if (days >= 1) {
    dayPart = `${days}d `;
    hoursPart = `${hours}h `;
  }
  return `${dayPart}${hoursPart}${minutes}m`;
}
