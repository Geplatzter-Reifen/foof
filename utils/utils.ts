export function timeout(ms: number) {
  return new Promise((resolve) => setTimeout(resolve, ms));
}

/**
 * Rounds a number to a given number of decimals.
 * @param decimals
 * @param number
 * @returns The rounded number or undefined if the input number is undefined.
 */
export function roundNumber(decimals: number, number?: number) {
  if (number === undefined) {
    return undefined;
  }
  return Math.round(number * Math.pow(10, decimals)) / Math.pow(10, decimals);
}
