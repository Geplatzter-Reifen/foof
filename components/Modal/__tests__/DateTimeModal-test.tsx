import { render, fireEvent } from "@/test-utils/test-utils";
import DateTimeModal from "@/components/Modal/DateTimeModal";
import { formatDate, DateFormat } from "@/utils/dateUtils";

describe("DateTimeModal Component", () => {
  const mockOnClose = jest.fn();
  const mockOnSave = jest.fn();
  const mockOnStartDateChange = jest.fn();
  const mockOnEndDateChange = jest.fn();

  const startDate = new Date(2024, 0, 1, 12, 0, 0);
  const endDate = new Date(2024, 0, 2, 13, 0, 0);

  const defaultProps = {
    modalVisible: true,
    onClose: mockOnClose,
    onSave: mockOnSave,
    onStartDateChange: mockOnStartDateChange,
    onEndDateChange: mockOnEndDateChange,
    initialStartDate: startDate,
    initialEndDate: endDate,
  };

  it("matches the snapshot", () => {
    const { toJSON } = render(<DateTimeModal {...defaultProps} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders correctly", () => {
    const { getByText } = render(<DateTimeModal {...defaultProps} />);
    // Startzeit
    expect(getByText("Startzeit")).toBeTruthy();
    expect(getByText(formatDate(startDate, DateFormat.DATE))).toBeTruthy();
    expect(getByText(formatDate(startDate, DateFormat.TIME))).toBeTruthy();
    // Endzeit
    expect(getByText("Endzeit")).toBeTruthy();
    expect(getByText(formatDate(endDate, DateFormat.DATE))).toBeTruthy();
    expect(getByText(formatDate(endDate, DateFormat.TIME))).toBeTruthy();
  });

  it("calls onClose when Abbrechen button is pressed", () => {
    const { getByText } = render(<DateTimeModal {...defaultProps} />);
    fireEvent.press(getByText("Abbrechen"));
    expect(mockOnClose).toHaveBeenCalled();
  });

  it("calls onSave when Speichern button is pressed", () => {
    const { getByText } = render(<DateTimeModal {...defaultProps} />);
    fireEvent.press(getByText("Speichern"));
    expect(mockOnSave).toHaveBeenCalled();
  });

  it("calls onStartDateChange with new date", () => {
    const newStartDate = new Date(2024, 0, 2);
    const { getByText, getByTestId } = render(
      <DateTimeModal {...defaultProps} />,
    );
    fireEvent.press(getByText(formatDate(startDate, DateFormat.DATE)));
    fireEvent(getByTestId("dateTimePicker"), "onChange", {
      nativeEvent: { timestamp: newStartDate.getTime() },
    });
    expect(mockOnStartDateChange).toHaveBeenCalledWith(newStartDate);
  });

  it("calls onEndDateChange with new time", () => {
    const newEndDate = new Date(2024, 0, 1, 14, 0, 0);
    const { getByText, getByTestId } = render(
      <DateTimeModal {...defaultProps} />,
    );
    fireEvent.press(getByText(formatDate(endDate, DateFormat.TIME)));
    fireEvent(getByTestId("dateTimePicker"), "onChange", {
      nativeEvent: { timestamp: newEndDate.getTime() },
    });
    expect(mockOnEndDateChange).toHaveBeenCalledWith(newEndDate);
  });
});
