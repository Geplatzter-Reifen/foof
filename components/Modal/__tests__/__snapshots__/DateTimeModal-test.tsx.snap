// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTimeModal Component matches the snapshot 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onDismiss={[MockFunction]}
  onRequestClose={[MockFunction]}
  statusBarTranslucent={true}
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "bottom": 0,
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      testID="@backdrop"
    />
    <View
      animationType="fade"
      backdropStyle={
        {
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
        }
      }
      onBackdropPress={[MockFunction]}
      onLayout={[Function]}
      shouldUseContainer={true}
      style={
        [
          {
            "width": "90%",
          },
          {
            "position": "absolute",
          },
          {
            "left": -999,
            "top": -999,
          },
        ]
      }
      visible={true}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#FFFFFF",
            "borderColor": "#7e7e7e",
            "borderRadius": 8,
            "borderWidth": 0,
            "elevation": 3,
            "opacity": 1,
            "overflow": "hidden",
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 2,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": undefined,
              "height": 0,
            }
          }
        />
        <View
          style={
            [
              {
                "flexShrink": 1,
              },
              {
                "paddingHorizontal": 24,
                "paddingVertical": 16,
              },
            ]
          }
        >
          <Text
            appearance="default"
            category="h5"
            status="basic"
            style={
              [
                {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 22,
                  "fontWeight": "800",
                },
                {
                  "textAlign": "center",
                },
              ]
            }
          >
            Startzeit
          </Text>
          <View
            appearance="default"
            level="1"
            style={
              [
                {
                  "backgroundColor": "#FFFFFF",
                },
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 5,
                },
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#FF6E14",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginHorizontal": 8,
                  "marginVertical": 3,
                  "minHeight": 40,
                  "minWidth": 40,
                  "opacity": 1,
                  "paddingHorizontal": 10,
                  "paddingVertical": 12,
                }
              }
            >
              <Text
                appearance="default"
                category="p1"
                status="basic"
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "fontWeight": "400",
                    },
                    {
                      "color": "#FF6E14",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "bold",
                      "marginHorizontal": 10,
                    },
                  ]
                }
              >
                01.01.2024
              </Text>
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={20}
                bbWidth={20}
                focusable={false}
                height={20}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "height": 20,
                      "marginHorizontal": 10,
                      "tintColor": "#FF6E14",
                      "width": 20,
                    },
                    {
                      "flex": 0,
                      "height": 20,
                      "width": 20,
                    },
                  ]
                }
                vbHeight={512}
                vbWidth={448}
                width={20}
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                >
                  <RNSVGPath
                    d="M96 32l0 32L48 64C21.5 64 0 85.5 0 112l0 48 448 0 0-48c0-26.5-21.5-48-48-48l-48 0 0-32c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 32L160 64l0-32c0-17.7-14.3-32-32-32S96 14.3 96 32zM448 192L0 192 0 464c0 26.5 21.5 48 48 48l352 0c26.5 0 48-21.5 48-48l0-272z"
                    fill={
                      {
                        "payload": 4294929940,
                        "type": 0,
                      }
                    }
                    propList={
                      [
                        "fill",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#FF6E14",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginHorizontal": 8,
                  "marginVertical": 3,
                  "minHeight": 40,
                  "minWidth": 40,
                  "opacity": 1,
                  "paddingHorizontal": 10,
                  "paddingVertical": 12,
                }
              }
            >
              <Text
                appearance="default"
                category="p1"
                status="basic"
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "fontWeight": "400",
                    },
                    {
                      "color": "#FF6E14",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "bold",
                      "marginHorizontal": 10,
                    },
                  ]
                }
              >
                12:00
              </Text>
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={20}
                bbWidth={20}
                focusable={false}
                height={20}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "height": 20,
                      "marginHorizontal": 10,
                      "tintColor": "#FF6E14",
                      "width": 20,
                    },
                    {
                      "flex": 0,
                      "height": 20,
                      "width": 20,
                    },
                  ]
                }
                vbHeight={512}
                vbWidth={512}
                width={20}
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                >
                  <RNSVGPath
                    d="M256 0a256 256 0 1 1 0 512A256 256 0 1 1 256 0zM232 120l0 136c0 8 4 15.5 10.7 20l96 64c11 7.4 25.9 4.4 33.3-6.7s4.4-25.9-6.7-33.3L280 243.2 280 120c0-13.3-10.7-24-24-24s-24 10.7-24 24z"
                    fill={
                      {
                        "payload": 4294929940,
                        "type": 0,
                      }
                    }
                    propList={
                      [
                        "fill",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
          <Text
            appearance="default"
            category="h5"
            status="basic"
            style={
              [
                {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 22,
                  "fontWeight": "800",
                },
                {
                  "textAlign": "center",
                },
              ]
            }
          >
            Endzeit
          </Text>
          <View
            appearance="default"
            level="1"
            style={
              [
                {
                  "backgroundColor": "#FFFFFF",
                },
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 5,
                },
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#FF6E14",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginHorizontal": 8,
                  "marginVertical": 3,
                  "minHeight": 40,
                  "minWidth": 40,
                  "opacity": 1,
                  "paddingHorizontal": 10,
                  "paddingVertical": 12,
                }
              }
            >
              <Text
                appearance="default"
                category="p1"
                status="basic"
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "fontWeight": "400",
                    },
                    {
                      "color": "#FF6E14",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "bold",
                      "marginHorizontal": 10,
                    },
                  ]
                }
              >
                02.01.2024
              </Text>
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={20}
                bbWidth={20}
                focusable={false}
                height={20}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "height": 20,
                      "marginHorizontal": 10,
                      "tintColor": "#FF6E14",
                      "width": 20,
                    },
                    {
                      "flex": 0,
                      "height": 20,
                      "width": 20,
                    },
                  ]
                }
                vbHeight={512}
                vbWidth={448}
                width={20}
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                >
                  <RNSVGPath
                    d="M96 32l0 32L48 64C21.5 64 0 85.5 0 112l0 48 448 0 0-48c0-26.5-21.5-48-48-48l-48 0 0-32c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 32L160 64l0-32c0-17.7-14.3-32-32-32S96 14.3 96 32zM448 192L0 192 0 464c0 26.5 21.5 48 48 48l352 0c26.5 0 48-21.5 48-48l0-272z"
                    fill={
                      {
                        "payload": 4294929940,
                        "type": 0,
                      }
                    }
                    propList={
                      [
                        "fill",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#FF6E14",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginHorizontal": 8,
                  "marginVertical": 3,
                  "minHeight": 40,
                  "minWidth": 40,
                  "opacity": 1,
                  "paddingHorizontal": 10,
                  "paddingVertical": 12,
                }
              }
            >
              <Text
                appearance="default"
                category="p1"
                status="basic"
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "fontWeight": "400",
                    },
                    {
                      "color": "#FF6E14",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "bold",
                      "marginHorizontal": 10,
                    },
                  ]
                }
              >
                13:00
              </Text>
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={20}
                bbWidth={20}
                focusable={false}
                height={20}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "height": 20,
                      "marginHorizontal": 10,
                      "tintColor": "#FF6E14",
                      "width": 20,
                    },
                    {
                      "flex": 0,
                      "height": 20,
                      "width": 20,
                    },
                  ]
                }
                vbHeight={512}
                vbWidth={512}
                width={20}
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                >
                  <RNSVGPath
                    d="M256 0a256 256 0 1 1 0 512A256 256 0 1 1 256 0zM232 120l0 136c0 8 4 15.5 10.7 20l96 64c11 7.4 25.9 4.4 33.3-6.7s4.4-25.9-6.7-33.3L280 243.2 280 120c0-13.3-10.7-24-24-24s-24 10.7-24 24z"
                    fill={
                      {
                        "payload": 4294929940,
                        "type": 0,
                      }
                    }
                    propList={
                      [
                        "fill",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
          <View
            appearance="default"
            level="1"
            style={
              [
                {
                  "backgroundColor": "#FFFFFF",
                },
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 5,
                },
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#FF6E14",
                  "borderColor": "#FF6E14",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginHorizontal": 4,
                  "marginTop": 5,
                  "minHeight": 40,
                  "minWidth": 40,
                  "opacity": 1,
                  "paddingHorizontal": 10,
                  "paddingVertical": 12,
                }
              }
            >
              <Text
                appearance="default"
                category="p1"
                status="basic"
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "fontWeight": "400",
                    },
                    {
                      "color": "#FFFFFF",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "bold",
                      "marginHorizontal": 10,
                    },
                  ]
                }
              >
                Abbrechen
              </Text>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#FF6E14",
                  "borderColor": "#FF6E14",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginHorizontal": 4,
                  "marginTop": 5,
                  "minHeight": 40,
                  "minWidth": 40,
                  "opacity": 1,
                  "paddingHorizontal": 10,
                  "paddingVertical": 12,
                }
              }
            >
              <Text
                appearance="default"
                category="p1"
                status="basic"
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "fontWeight": "400",
                    },
                    {
                      "color": "#FFFFFF",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "bold",
                      "marginHorizontal": 10,
                    },
                  ]
                }
              >
                Speichern
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
